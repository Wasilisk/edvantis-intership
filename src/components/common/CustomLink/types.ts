export type CustomLinkProps = {
    to: string,
    label: string
}