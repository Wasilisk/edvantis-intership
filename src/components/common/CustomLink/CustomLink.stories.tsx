import {Story} from "@storybook/react";
import React from "react";
import CustomLink from "./index";
import {CustomLinkProps} from "./types";

export default {
    title: 'Application/Links/CustomLink',
    component: CustomLink,
};

const Template: Story<CustomLinkProps> = (props) => <CustomLink {...props}/>;

export const Default = Template.bind({});
Default.args = {
    label: "Link",
    to: "/link/to"
};