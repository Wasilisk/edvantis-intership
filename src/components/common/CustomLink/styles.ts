import {Link} from "react-router-dom";
import styled from "styled-components";

export const CustomLinkStyles = styled(Link)`
  width: fit-content;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: all 0.2s;
  color: #242424;
  font-size: 12px;
  font-family: 'Hind', sans-serif;
  letter-spacing: 0;
  font-weight: 600;
  text-transform: uppercase;
  text-decoration: none;
  
  svg {
    height: 24px;
    width: 24px;
    margin-left: 5px;
  }
  
  :hover {
    text-decoration: underline;
    text-decoration-color: black;
  }
`