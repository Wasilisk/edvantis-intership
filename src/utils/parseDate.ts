const month = ["January","February","March","April","May","June","July","August","September","October","November","December"];

export const parseDate = (timestamp: number) => {
    const date = new Date(timestamp);

    return `${month[date.getMonth()]} ${('0' + date.getDate()).slice(-2)}, ${date.getFullYear()}`;
}