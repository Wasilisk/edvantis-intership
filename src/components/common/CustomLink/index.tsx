import React from 'react';
import {CustomLinkStyles} from "./styles";
import {CustomLinkProps} from "./types";
import ButtonArrow from "../../../assets/icons/ArrowIcon";

const CustomLink = ({to, label}: CustomLinkProps) => {
    return (
        <CustomLinkStyles to={to}>
            {label} <ButtonArrow/>
        </CustomLinkStyles>
    );
};


export default CustomLink;