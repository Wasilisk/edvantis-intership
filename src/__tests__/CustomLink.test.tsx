import {render} from "./utils/store-wrapper";
import CustomLink from "../components/common/CustomLink";
import {fireEvent} from "@testing-library/react";
import {mockHistory} from "./utils/history-wrapper";

describe('CustomLink component', () => {
    it('should render component', () => {
        const {getByText, asFragment} = render(<CustomLink to="/link" label="Link"/>);

        const linkLabel = getByText("Link");
        expect(linkLabel).toBeDefined();

        fireEvent.click(linkLabel);
        expect(mockHistory.location.pathname).toBe("/link");

        expect(asFragment()).toMatchSnapshot();
    })
})