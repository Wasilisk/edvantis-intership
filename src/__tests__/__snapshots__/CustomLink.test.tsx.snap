// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLink component should render component 1`] = `
<DocumentFragment>
  <a
    class="sc-bczRLJ jZFBMR"
    href="/link"
  >
    Link 
    <svg
      height="100%"
      viewBox="0 0 36 36"
      width="100%"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        class="cls-1"
        d="M34,18l-5-4v3H26v2h3v3ZM24,19H20V17h4v2Zm-6,0H14V17h4v2Zm-6,0H8V17h4v2ZM4,16a2,2,0,1,1-2,2A2,2,0,0,1,4,16Z"
        data-name="28"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;
